### PR TITLE
Github actions - debian builds

### PR DESCRIPTION
**Pull request by MateuszStudiosCompany**
- Move build process into `.github/actions/build/action.yml`
  - Less code duplication
  - `make build` instead of `make install`
  - Add `libgirepository1.0-dev` into installed dependencies (i cannot build PyGObject without it)
  - Add param `target-distro` which determinate if `Prepare deb build (for Debian)` step will run
  - Add job artifacts upload (`.deb` files)
- Update build and upload jobs to use `.github/actions/build/action.yml`
- Build job will fire for every branch instead of `master`
- Update readme with debian deps
- Update .gitignore with `.vscode/` dir

### CODE REVIEW

File: .github/actions/build/action.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+name: "Build Webgreeter"
+description: "Builds webgreeter"
```
Comment (JezerM) [inline .github/actions/build/action.yml:2]:
```suggestion
name: "Build web-greeter"
description: "Builds web-greeter"
```

File: .github/actions/build/action.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+     description: "Target distro"
+runs:
+  using: "composite"
+  steps:
+    -
+      name: Checkout
+      uses: actions/checkout@v2
+    -
+      name: Install dependencies
+      shell: bash
```
Comment (JezerM) [inline .github/actions/build/action.yml:15]: Is `shell: bash` required? I guess the default one is enough.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -1,26 +1,27 @@
 name: Build Test
 
-on:
-  push:
-    branches: [ master ]
-  pull_request:
-    branches: [ master ]
+on: [push]
```
Comment (JezerM) [inline .github/workflows/build.yml:3]:
```suggestion
on: [push, pull_request]
```

Build tests should also cover PRs.

File: .github/actions/build/action.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        cd build/
+        dpkg-deb --root-owner-group --build install_root "web-greeter.deb"
+        dpkg --info "./web-greeter.deb"
+    -
+      name: 'Upload Artifact'
+      uses: actions/upload-artifact@v2
+      with:
+        name: build-${{ inputs.target-distro }}
+        path: ./build/web-greeter.deb
+        retention-days: 7
```
Comment (JezerM) [inline .github/actions/build/action.yml:58]: Why is this necessary if there's no `actions/download-artifact@v2` in publish?

**Review by JezerM — CHANGES_REQUESTED** [review]
I don't know much about GitHub workflows and actions, but overall seems good~

File: .github/actions/build/action.yml (hunk 2 above)
Comment (MateuszStudiosCompany) [reply .github/actions/build/action.yml:15]: Looks like `shell` is required here for some reason. Without it i got `Error: /home/runner/work/web-greeter/web-greeter/./.github/actions/build/action.yml (Line: 14, Col: 7): Required property is missing: shell`. I also don't know much about actions so maybe i missed someting :o

File: .github/actions/build/action.yml (hunk 4 above)
Comment (MateuszStudiosCompany) [reply .github/actions/build/action.yml:58]: I forgot to mention it. I added artifacts upload to be easily accessible to download it after every job from "run" page (ex. https://github.com/MateuszStudiosCompany/web-greeter/actions/runs/1815314221). I personally use them for testing/debugging. So this is totally optionall and we cen remove it if you want.

File: .github/actions/build/action.yml (hunk 2 above)
Comment (JezerM) [reply .github/actions/build/action.yml:15]: Well, it's not a problem, just curious~

File: .github/actions/build/action.yml (hunk 4 above)
Comment (JezerM) [reply .github/actions/build/action.yml:58]: Then, no problem~

**Review by JezerM — APPROVED** [review]
Looks good! Thanks for these changes~
